File: Practice/20/C++/code.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

int price, volue;
char name;
struct napitock{
    string name;
    long long price = 0;
    long long volue = 0;
};

int main()
{
    napitock poidet_v_sitopiano;
    napitock vvod;
    long long deneg, litrov, luche_litrov, skolko_napitkov, k = 0;
    cin >> deneg >> skolko_napitkov;
    for (int i = 0; i < skolko_napitkov; i++)
    {
        cin >> vvod.name >> vvod.price >> vvod.volue;
        litrov = (deneg/vvod.price) * vvod.volue;
        if (litrov == 0) continue;
        if (k == 0) poidet_v_sitopiano = vvod; k+=1;
        luche_litrov = (deneg/poidet_v_sitopiano.price) * poidet_v_sitopiano.volue;
        if (litrov > luche_litrov) poidet_v_sitopiano = vvod;
    }
    if (poidet_v_sitopiano.price == 0) cout << "Error" << endl;
    else 
    {
        cout << poidet_v_sitopiano.name << ' ' << deneg / poidet_v_sitopiano.price << endl; 
        cout << deneg / poidet_v_sitopiano.price * poidet_v_sitopiano.volue << endl;
        cout << deneg - poidet_v_sitopiano.price * (deneg / poidet_v_sitopiano.price) << endl;
    }
}

File: Practice/20/python/code.py
class napitok:
    def __init__(self):
        self.name = ''
        self.price = 0.0
        self.voluem = 0.0
deneg = int(input())
kovlo_napitkov = int(input())
chto_poidet_na_buhich = napitok()
vvod_nap = napitok()
k = 0
for i in range (kovlo_napitkov):
    a, b, c = map(str, input().split(' '))
    b = int(b)
    c = int(c)
    a1,b1,c1 = a,b,c
    vvod_nap.name, vvod_nap.price, vvod_nap.voluem = a, b, c
    litrov = (deneg // vvod_nap.price) * vvod_nap.voluem
    if litrov == 0:
        continue
    if k == 0:
        chto_poidet_na_buhich.name, chto_poidet_na_buhich.price, chto_poidet_na_buhich.voluem = a1,b1,c1
        k += 1
    skolko_litrov_luchshe = (deneg // chto_poidet_na_buhich.price) * chto_poidet_na_buhich.voluem
    if litrov > skolko_litrov_luchshe:
        chto_poidet_na_buhich = vvod_nap
if chto_poidet_na_buhich.price == 0:
    print('-1')
else:
    print(chto_poidet_na_buhich.name, deneg // chto_poidet_na_buhich.price)
    print(deneg // chto_poidet_na_buhich.price * chto_poidet_na_buhich.voluem)
    print(deneg - chto_poidet_na_buhich.price * (deneg // chto_poidet_na_buhich.price))